commons/zenoh-shm/src/api/provider/shared_memory_provider.rs

```diff
@@ -0,0 +1,910 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use std::{
+    collections::VecDeque,
+    marker::PhantomData,
+    sync::{atomic::Ordering, Arc, Mutex},
+    time::Duration,
+};
+
+use async_trait::async_trait;
+use zenoh_result::ZResult;
+
+use crate::{
+    api::{common::types::ProtocolID, slice::zsliceshmmut::ZSliceShmMut},
+    header::{
+        allocated_descriptor::AllocatedHeaderDescriptor, descriptor::HeaderDescriptor,
+        storage::GLOBAL_HEADER_STORAGE,
+    },
+    watchdog::{
+        allocated_watchdog::AllocatedWatchdog,
+        confirmator::{ConfirmedDescriptor, GLOBAL_CONFIRMATOR},
+        descriptor::Descriptor,
+        storage::GLOBAL_STORAGE,
+        validator::GLOBAL_VALIDATOR,
+    },
+    SharedMemoryBuf, SharedMemoryBufInfo,
+};
+
+use super::{
+    chunk::{AllocatedChunk, ChunkDescriptor},
+    shared_memory_provider_backend::SharedMemoryProviderBackend,
+    types::{AllocAlignment, BufAllocResult, ChunkAllocResult, MemoryLayout, ZAllocError},
+};
+
```

Comment:
```suggestion

#[zenoh_macros::unstable_doc]
```

Reply:
BusyChunk should not be pub at all, so I remove it from public